example/lib/dog_mapping/dog_mapping.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:smartstruct/smartstruct.dart';
+part 'dog_mapping.mapper.g.dart';
+
+// TARGET
+
+class Dog {
+  final String name;
+  final String breed;
+  final int age;
+
+  Dog(this.name, this.breed, this.age);
+}
+
+// SOURCE
+
+class DogModel {
+  final String dogName;
+  final String breed;
+
+  DogModel(
+    this.dogName,
+    this.breed,
+  );
+}
+
+/// Mapper showcasing explicit fieldmapping in case fields do not match their respective fieldnames
+@Mapper()
+abstract class DogMapper {
+  @Mapping(source: 'dogName', target: 'name')
+  @Mapping(source: 'age', target: 'age', isStraight: true)
```

Comment:
I really like the idea of this functionality. I am not sure if we need an extra configuration though.
Wouldn't it be nicer if the mapper can map these kinds of extra arguments straight away? Without extra configuration? Or am I missing some edge case.

Reply:
I think about if we need the extra configuration. I find if we don't use the **isStraight** attribute, this mapping relation will be considered as the inner attribute of the object corresponding to the attribute of the target object in the origin way. 
So, there will be a problem if the method parameter name is same as the field name which in the old object, we won't choose to use which one to fill the target object. Both data will be possible.
We can solve the problem by defining the priority of two data. For example, the method parameter will be preferentially selected. But the user will can't choose the inner field. I want to keep the flexibility for user and keep all origin use methods.

Reply:
So I was thinking something like this https://mapstruct.org/documentation/stable/reference/html/#mappings-with-several-source-parameters
See Example 11. Mapping method directly referring to a source parameter
`@Mapper
public interface AddressMapper {

    @Mapping(target = "description", source = "person.description")
    @Mapping(target = "houseNumber", source = "hn")
    DeliveryAddressDto personAndAddressToDeliveryAddressDto(Person person, Integer hn);
}`
This would require some refactoring though, because currently if we write something like this "hn", the Generator expects also a source parameter to access "hn" from.
So for now the extra parameter should be the best solution, and maybe at a later time we can build it, so the parameter will be implied.

Reply:
Yes, I was inspired by the original `mapstruct`. I think that is a better way, but the code needs to be refactored. Based on the current situation, I chose to add a parameter. I think that I don't have the ability to refactor the code. If you can complete the refactoring, it will be better.